dev/AppLifecycle/ActivationRegistrationManager.cpp

```diff
@@ -8,47 +8,89 @@
 #include "ProtocolActivatedEventArgs.h"
 #include "FileActivatedEventArgs.h"
 #include "Association.h"
+#include "ExtensionContract.h"
+#include "EncodedLaunchExecuteCommand.h"
+
+extern "C" HRESULT __stdcall DllRegisterServer() noexcept;
 
 namespace winrt::Microsoft::ProjectReunion::implementation
 {
+    std::wstring GenerateEncodedLaunchUri(std::wstring const& appUserModelId, std::wstring const& contractId)
```

Comment:
>GenerateEncodedLaunchUri [](start = 17, length = 24)

This assumes valid inputs e.g. can't be "It's not a valid AUMID & other <badness> that's not (properly) eñcod=d!" but I don't see where that's prevented e.g. RegisterForFileTypeActivation accepts strings with no validation which are then encoded as-is in the URI. Somewhere in that chain the parameters should be (1) validated and (2) URI escaped.

Reply:
Yes, that is done as part of the Uri class.

Reply:
URI escaping is handled in the Uri class as part of creating the arguments.  Encoding it early doesn't gain any validation as there aren't any wrong data options that cause validation to fail.  Everything is encoded with escape sequences regardless of the data.

The best that could be done would be to ensure the combination of everything is still a valid Uri before progressing, which is what I'll probably add for that here.  All that ensures is that the code in this component is correct given the two inputs which of course is good.